File: test/corpus/go/duplicates.go
package corpus

// An exact clone pair: identical up to renamed identifiers, above the clone size
// floor.

// dendro-expect: duplicate
func totalPrice(items []int, tax int) int {
    subtotal := 0
    for i := 0; i < len(items); i++ {
        subtotal = subtotal + items[i]
    }
    grand := subtotal + subtotal*tax
    rounded := roundValue(grand)
    return rounded
}

// dendro-expect: duplicate
func finalAmount(rows []int, rate int) int {
    accum := 0
    for i := 0; i < len(rows); i++ {
        accum = accum + rows[i]
    }
    whole := accum + accum*rate
    snapped := roundValue(whole)
    return snapped
}

File: test/corpus/javascript/duplicates.js
// An exact clone pair: identical up to renamed identifiers, above the clone size
// floor.

// dendro-expect: duplicate
function totalPrice(items, tax) {
    let subtotal = 0;
    for (let i = 0; i < items.length; i++) {
        subtotal = subtotal + items[i];
    }
    let grand = subtotal + subtotal * tax;
    let rounded = roundValue(grand);
    return rounded;
}

// dendro-expect: duplicate
function finalAmount(rows, rate) {
    let accum = 0;
    for (let i = 0; i < rows.length; i++) {
        accum = accum + rows[i];
    }
    let whole = accum + accum * rate;
    let snapped = roundValue(whole);
    return snapped;
}
